presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -73,6 +74,42 @@ public static GeometryTypeName valueOf(String type)
         }
     }
 
+    public static Envelope deserializeEnvelope(Slice shape)
```

Comment:
Would it be possible that `Slice shape` contains raw data for envelope. It seems that `Envelope#merge(com.esri.core.geometry.Envelope)` might have quadratic complexity.

Reply:
Could you add benchmark of this new method?

Reply:
@sopel39 , at the moment we have little control over serde of the geometry type, because we use serde from the ESRI library. I do keep running into cases where custom serde fully controlled in Presto would allow for more optimizations, but for now I want to keep it as is.  `Slice shape` contains raw data for the envelope for all shape types except for geometry collection which is rarely used.